src/object/cli_obj.c

```diff
@@ -3626,6 +3813,13 @@ obj_shard_comp_cb(struct shard_auxi_args *shard_auxi,
 	if (ret == 0) {
 		if (obj_auxi->map_ver_reply < shard_auxi->map_ver)
 			obj_auxi->map_ver_reply = shard_auxi->map_ver;
+		if (obj_req_is_ec_cond_fetch(obj_auxi)) {
+			iter_arg->cond_fetch_exist = true;
+			if (obj_auxi->result == -DER_NONEXIST)
+				obj_auxi->result = 0;
```

Comment:
here the shard replied ret == 0 it means the dkey/akey exist, "obj_auxi->result == -DER_NONEXIST" means some other shards returned NONEXIST before, so in this case should zero the rc because any shard return exist then it is exist.